File: src/main/java/org/usfirst/frc/team3863/commands/BaseCommand.java
package org.usfirst.frc.team3863.commands;

import edu.wpi.first.wpilibj.command.Command;
import org.usfirst.frc.team3863.OI;
import org.usfirst.frc.team3863.subsystems.*;

/**
 * Created by robotics on 2/9/2016.
 */
public abstract class BaseCommand extends Command {

    public static OI oi;

    //SubSystems
    public static Pneumatics pneumatics = new Pneumatics();
    public static DriveTrain driveTrain = new DriveTrain();
    public static Arm arm = new Arm();
    public static Shooter shooter = new Shooter();
    public static Intake intake = new Intake();

    public static void init() {
        oi = new OI();
    }

    public BaseCommand() {
        super();
    }

    public BaseCommand(String name) {
        super(name);
    }

    public BaseCommand(String name, double timeout) {
        super(name, timeout);
    }

    public BaseCommand(double timeout) {
        super(timeout);
    }
}

File: src/main/kotlin/org/usfirst/frc/team3863/kotlin/commands/BaseCommand.kt
package org.usfirst.frc.team3863.kotlin.commands

import edu.wpi.first.wpilibj.command.Command
import org.usfirst.frc.team3863.kotlin.OI
import org.usfirst.frc.team3863.kotlin.subsystems.Arm
import org.usfirst.frc.team3863.kotlin.subsystems.DriveTrain
import org.usfirst.frc.team3863.kotlin.subsystems.Pneumatics

/**
 * Created by Joshua Freedman on 2/11/2016.
 * Project: 2016Robot
 */
abstract class BaseCommand : Command {

    constructor(name: String) : super(name) {
    }

    companion object {

        var oi: OI = OI()

        //SubSystems
        var pneumatics = Pneumatics()
        var driveTrain = DriveTrain()
        var arm = Arm()
    }
}
